api/illumio/cloud/k8sclustersync/v1/k8s_info.proto

```diff
@@ -0,0 +1,327 @@
+    KubernetesResourceSnapshotComplete resource_snapshot_complete = 3;
+    // A mutation to a Kubernetes resource in the cluster.
+    KubernetesResourceMutation kubernetes_resource_mutation = 4;
+  }
+}
+
+// Empty on purpose since this type is just expected when snapshot stream is done.
+message KubernetesResourceSnapshotComplete {}
+
+// Message sent to the operator in a KubernetesResources response stream.
+message SendKubernetesResourcesResponse {
+  // oneof response {
+  //   // Until we are writing policies we shouldnt worry about what is returned here
+  //   KubernetesAck ack = 1;
+  // }
+}
+
+// A mutation to a Kubernetes resource in the cluster.
+// Each mutation contains the full resource metadata after mutation (on create or update)
+// or before mutation (on delete).
+message KubernetesResourceMutation {
+  // Send any of these "mutation" messages that can contain any of
+  // The documented types
+  oneof mutation {
+    // Metadata of a newly created Kubernetes resource.
+    KubernetesObjectMetadata create_resource = 1;
+    // Metadata of an updated Kubernetes resource, after update.
+    KubernetesObjectMetadata update_resource = 2;
+    // Metadata of a deleted Kubernetes resource, before deletion.
+    KubernetesObjectMetadata delete_resource = 3;
+  }
+}
+
+// A flow received from Cilium Hubble Relay using the Observer service's GetFlows RPC.
+// https://github.com/cilium/cilium/blob/main/api/v1/observer/observer.proto
+message CiliumFlow {
+  // Time when flow occured.
+  google.protobuf.Timestamp time = 1;
+  // node_name is the name of the node from which this Flow was captured.
+  string node_name = 2;
+  // traffic_direction distinguishes direction of the connection, e.g. ingress or egress.
+  TrafficDirection traffic_direction = 3;
+  // verdict of the flow, indicating the outcome of the network packet processing.
+  Verdict verdict = 4;
+  // layer3 IP layer, source, destination and ip_version
+  IP layer3 = 5;
+  // layer4 contains protocol used
+  Layer4 layer4 = 6;
+  // endpoint of source ip with metadata
+  optional Endpoint source_endpoint = 9;
+  // endpoint of destination ip with metadata
+  optional Endpoint destination_endpoint = 10;
+  // destination_service contains the service name of the destination
+  Service destination_service = 11;
+  // The CiliumNetworkPolicies allowing the egress of the flow.
+  repeated Policy egress_allowed_by = 13;
+  // The CiliumNetworkPolicies allowing the ingress of the flow.
+  repeated Policy ingress_allowed_by = 14;
+  // The CiliumNetworkPolicies denying the egress of the flow.
+  repeated Policy egress_denied_by = 15;
+  // The CiliumNetworkPolicies denying the ingress of the flow.
+  repeated Policy ingress_denied_by = 16;
+  // is_reply indicates that this was a packet (L4) or message (L7) in the
+  // reply direction. May be absent (in which case it is unknown whether it
+  // is a reply or not).
+  google.protobuf.BoolValue is_reply = 17;
```

Comment:
Maybe move this up to tag 7, to not waste it. :)

Reply:
Oh my god how did I miss 7 AND 8. okay fixed this.